ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,563 @@
+
+    return obj_set
+
+
+def measure_attach_time(pod_objs, pvc_size):
+    """
+    Measures and Logs Attach Time of all PODs.
+
+    Args:
+        pod_objs (list) : List of POD objects for which we have to measure the time.
+        pvc_size (int) : Size of each PVC in GB.
+
+    Raises:
+        PerformanceException : Raises an exception if POD attach time is greater than the accepted time.
+
+    Logs:
+        Attach time of all PODs, as well as the average time.
+
+    """
+    msg_prefix = f"PVC size: {pvc_size}."
+    start_time_dict_list = []
+    for pod in pod_objs:
+        start_time_dict_list.append(helpers.pod_start_time(pod))
+    logger.info(str(start_time_dict_list))
+    time_measures = []
+    for attach_time in start_time_dict_list:
+        if "my-container" in attach_time:
+            time_measures.append(attach_time["my-container"])
+        else:
+            time_measures.append(attach_time["web-server"])
+    for index, start_time in enumerate(time_measures):
+        logger.info(
+            f"{msg_prefix} pod number {index + 1} start time: {start_time} seconds"
+        )
+        if start_time > 30:
+            raise ex.PerformanceException(
+                f"{msg_prefix} Pod number {index} start time is {start_time},"
+                f"which is greater than 30 seconds"
+            )
+    if time_measures:
+        average = statistics.mean(time_measures)
+        logger.info(
+            f"{msg_prefix} The average time for the sampled {len(time_measures)} pods is {average} seconds."
+        )
+
+
+def measure_pod_creation_time(namespace, num_of_pods):
+    """
+    Measures and Logs the POD Creation Time of all the PODs.
+
+    Args:
+        namespace (str) : Namespace in which the PODs are created.
+        num_of_pods (int) : Number of PODs created.
+
+    Raises:
+        PerformanceException : Raises an exception if POD creation time is greater than the accepted time.
+
+    Logs:
+        POD Creation Time of all the PODs.
+
+    """
+    logs = performance_lib.run_oc_command(
+        "get events --sort-by='{.lastTimestamp}'",
+        namespace,
+    )
+
+    scheduled_time = None
+    pod_no = num_of_pods
+    accepted_creation_time = 12
+
+    for line in logs:
+        logger.info(line)
+        if "Scheduled" in line:
+            scheduled_time = int(line.split()[0][:-1])
+        elif "Created" in line:
+            created_time = int(line.split()[0][:-1])
+            creation_time = scheduled_time - created_time
+            logger.info(f"POD number {pod_no} was created in {creation_time} seconds.")
+            if creation_time > accepted_creation_time:
+                raise ex.PerformanceException(
+                    f"POD creation time is {creation_time} and is greater than "
+                    f"{accepted_creation_time} seconds."
+                )
+            pod_no -= 1
+
+
+def measure_pvc_creation_time(interface, pvc_size, pvc_objs, start_time):
```

Comment:
AFAIK PVC creation time does not depend on the size of the pvc, do we really need to pass the pvc_size here?

Reply:
No, we don't really need it, it was just used for logging here. It seems better to remove it.